test/integration/scheduler/podsready/scheduler_test.go

```diff
@@ -249,47 +249,63 @@ var _ = ginkgo.Describe("SchedulerWithWaitForPodsReady", func() {
+				Count: ptr.To[int32](2),
+			}, false)
+
+			ginkgo.By("the workload exceeded re-queue backoff limit should be deactivated")
 			util.ExpectWorkloadsToHaveQuotaReservation(ctx, k8sClient, prodClusterQ.Name, prodWl)
 			util.ExpectQuotaReservedWorkloadsTotalMetric(prodClusterQ, 3)
 			util.ExpectAdmittedWorkloadsTotalMetric(prodClusterQ, 3)
 			time.Sleep(podsReadyTimeout)
-			ginkgo.By("evicted re-admitted workload should have 2 in the re-queue count")
-			util.ExpectWorkloadToHaveRequeueCount(ctx, k8sClient, client.ObjectKeyFromObject(prodWl), ptr.To[int32](2))
-			ginkgo.By("the workload exceeded re-queue backoff limit should be deactivated")
 			gomega.Eventually(func(g gomega.Gomega) {
 				g.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(prodWl), prodWl)).Should(gomega.Succeed())
 				g.Expect(ptr.Deref(prodWl.Spec.Active, true)).Should(gomega.BeFalse())
+				g.Expect(prodWl.Status.RequeueState).Should(gomega.BeNil())
+				workload.SetRequeuedCondition(prodWl, kueue.WorkloadEvictedByDeactivation, "by test", false)
+				g.Expect(workload.ApplyAdmissionStatus(ctx, k8sClient, prodWl, true)).Should(gomega.Succeed())
 			}, util.Timeout, util.Interval).Should(gomega.Succeed())
-
 			// should observe a metrics of WorkloadEvictedByDeactivation
 			util.ExpectEvictedWorkloadsTotalMetric(prodClusterQ.Name, kueue.WorkloadEvictedByDeactivation, 1)
 
-			ginkgo.By("verify the re-activated inactive 'prod' workload re-queue state is reset")
+			ginkgo.By("the reactivated workload should not be deactivated by the scheduler unless exceeding the backoffLimitCount")
 			gomega.Eventually(func(g gomega.Gomega) {
 				g.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(prodWl), prodWl)).Should(gomega.Succeed())
 				prodWl.Spec.Active = ptr.To(true)
 				g.Expect(k8sClient.Update(ctx, prodWl)).Should(gomega.Succeed())
-			}, util.Timeout, util.Interval).Should(gomega.Succeed(), "Reactivate inactive Workload")
-			gomega.Eventually(func(g gomega.Gomega) {
-				g.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(prodWl), prodWl)).Should(gomega.Succeed())
-				g.Expect(prodWl.Status.RequeueState).Should(gomega.BeNil())
 			}, util.Timeout, util.Interval).Should(gomega.Succeed())
+			util.FinishEvictionForWorkloads(ctx, k8sClient, prodWl)
+			util.ExpectWorkloadsToHaveQuotaReservation(ctx, k8sClient, prodClusterQ.Name, prodWl)
+			util.ExpectQuotaReservedWorkloadsTotalMetric(prodClusterQ, 4)
+			util.ExpectAdmittedWorkloadsTotalMetric(prodClusterQ, 4)
+			util.AwaitWorkloadEvictionByPodsReadyTimeout(ctx, k8sClient, client.ObjectKeyFromObject(prodWl), podsReadyTimeout)
+			util.SetRequeuedConditionWithPodsReadyTimeout(ctx, k8sClient, client.ObjectKeyFromObject(prodWl))
+			util.FinishEvictionForWorkloads(ctx, k8sClient, prodWl)
+			util.ExpectWorkloadToHaveRequeueState(ctx, k8sClient, client.ObjectKeyFromObject(prodWl), &kueue.RequeueState{
+				Count: ptr.To[int32](1),
+			}, false)
```

Comment:
No strong view, but it feels there is quite a potential for race conditions and flakes here. 

Could we just end the test on the previous check, that the `RequeueState` is nil, and the metric is incremented?

Reply:
> No strong view, but it feels there is quite a potential for race conditions and flakes here.

I'm sure your concerns, but I think we need to have reactivation test to observe such https://github.com/kubernetes-sigs/kueue/issues/2174 bug.

So, let's replace the line 306-308 with verification of `spec.active=true`.
WDYT?

Reply:
> I'm sure your concerns, but I think we need to have reactivation test to observe such https://github.com/kubernetes-sigs/kueue/issues/2174 bug.

That is true, but since we now check that `requeueState` is cleaned, I'm not sure how a similar bug could happen, maybe if this was stored in-memory, but I suppose the risk is minimal.

> So, let's replace the line 306-308 with verification of spec.active=true.

Testing for `spec.active=true` would be better for sure.

But also `ExpectWorkloadsToHaveQuotaReservation` is time-sensitive and has a potential for raciness, because the workload gets evicted soon later `AwaitWorkloadEvictionByPodsReadyTimeout`. Maybe just check that the metrics were bumped  - this approach would not be time-sensitive.

Reply:
@mimowo I think that there is no such race condition since this integration test doesn't have jobframework reconciler, which means the Job is never readmitted unless performing the `FinishEvictionForWorkloads`, right?

Reply:
Ah, got it. So if the PodsReady timeout happens quickly then you would still pass the `ExpectWorkloadsToHaveQuotaReservation` (L300) check, because it is not removed until you call `FinishEvictionForWorkloads`. It convinces me there is no race condition indeed.

Reply:
Yes, exactly. I agree that comment would be helpful.